feature_tests/js/api/RenamedOpaqueIterable.mjs

```diff
@@ -0,0 +1,48 @@
+// generated by diplomat-tool
+import { RenamedOpaqueIterator } from "./RenamedOpaqueIterator.mjs"
+import wasm from "./diplomat-wasm.mjs";
+import * as diplomatRuntime from "./diplomat-runtime.mjs";
+
+
+const RenamedOpaqueIterable_box_destroy_registry = new FinalizationRegistry((ptr) => {
+    wasm.namespace_OpaqueIterable_destroy(ptr);
+});
+export class RenamedOpaqueIterable {
+    // Internal ptr reference:
+    #ptr = null;
+
+    // Lifetimes are only to keep dependencies alive.
+    // Since JS won't garbage collect until there are no incoming edges.
+    #selfEdge = [];
+    
+    
+    constructor(ptr, selfEdge) {
+        
+        this.#ptr = ptr;
+        this.#selfEdge = selfEdge;
+        // Unconditionally register to destroy when this object is ready to garbage collect.
+        RenamedOpaqueIterable_box_destroy_registry.register(this, this.#ptr);
+    }
+
+    get ffiValue() {
+        return this.#ptr;
+    }
+
+
+    [Symbol.iterator]() {
+        
+        // This lifetime edge depends on lifetimes 'a
+        let aEdges = [this];
+        const result = wasm.namespace_OpaqueIterable_iter(this.ffiValue);
```

Comment:
thought: we could use a separate private method here as well, but it's unclear to me if that's actually better. fine as is.